spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/RoutingKeyGenerator.scala

```diff
@@ -44,8 +44,12 @@ class RoutingKeyGenerator(table: TableDef, columnNames: Seq[String])
 
   private def fillRoutingKey(stmt: BoundStatement): Array[ByteBuffer] = {
     val rk = routingKey.get
-    for (i <- 0 until partitionKeyIdxs.size)
+    for (i ← partitionKeyIdxs.indices) {
+      if (stmt.isNull(partitionKeyIdxs(i)))
+        throw new NullPointerException(
```

Comment:
I can change to IAE. However, how introducing a custom exception would allow me to verify the exception message which should say which column is null?